src/ts/HttpReader.ts

```diff
@@ -266,13 +340,40 @@ class HttpRangeClient {
         this.bytesEverRequested += length;
 
         const range = `bytes=${begin}-${begin + length - 1}`;
-        Logger.debug(
+        Logger.info(
             `request: #${this.requestsEverMade}, purpose: ${purpose}), bytes: (this_request: ${length}, ever: ${this.bytesEverRequested}), Range: ${range}`
         );
 
         const response = await fetch(this.url, {
             headers: {
                 Range: range,
+                // TODO: better parallelize requests on Chrome
```

Comment:
This is really unfortunate. 

![image](https://user-images.githubusercontent.com/217057/122309712-9fceb200-cec3-11eb-98f4-9690c52526da.png)

Overall I think the changes are still worthwhile, even for chrome.

1. batching reduces number of requests and overall request size
2. Though it's significantly worse than Safari/Chrome, there is still *some* concurrency with Chrome - the second request starts as soon as we have the headers for the first. 

This example shows some longer requests - during which Chrome can benefit from concurrent feature fetching a bit more - notice how the 4th/5th request complete during the time the 3rd larger request is downloading.

![image](https://user-images.githubusercontent.com/217057/122310192-9f82e680-cec4-11eb-9e55-a1e1b544025a.png)

Reply:
Curious...I wonder if we can expect Chrome to fix this eventually? Seems like a rather big dent in it's performance track record.

Reply:
In case you missed it from the comment, this [https://bugs.chromium.org/p/chromium/issues/detail?id=969828](chrome bug), was filed two years ago.

My impression is that Chrome (and probably other folks) primarily think of range requests as being for:

- resuming broken downloads
- streaming/seeking through media

So our use case is pretty niche. The only other people apart from us that I'm aware of using this are bit torrent  implementations (webtorrent). I've also seen parallel Range requests used for privacy purposes (occluding from the outside network what content is being requested).

I don't have any way of knowing, but I don't expect it to be fixed soon.

Reply:
A humble plea to "star" the issue if you have a google account: https://bugs.chromium.org/p/chromium/issues/detail?id=969828

Reply:
done :)